### DISCUSSION TITLE
Merge confidence badges broken when grouping digest updates with others

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#27407] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

When a change set includes both version and digest updates (yes, I know this is not recommended :smile:) the merge confidence badges don't render properly:

https://github.com/fgreinacher/renovate-digest-confidence/pull/4

<img width="798" alt="image" src="https://github.com/renovatebot/renovate/assets/1540469/e502f450-5e51-4aa7-a04b-1ddb5f7c2ed2">

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by fgreinacher** [comment]
Root cause seems to be that the badge API cannot serve URLs without version (in the example above `https://developer.mend.io/api/mc/badges/age/docker/node/?slim=true`).

I see two directions to fix this:

1. adapt the badge API to render an `?` badge when no version is provided 
2. adapt the PR body definitions to render nothing when the version is missing

**Reply by rarkins** [reply]
@nabeelsaabna @Gabriel-Ladzaretti wdyt?

**Reply by nabeelsaabna** [reply]
we don't support docker images 
please check: https://docs.renovatebot.com/merge-confidence/#supported-languages

**Reply by rarkins** [reply]
@fgreinacher do you know how/why docker datasource has badges in your example?

**Reply by fgreinacher** [reply]
Yes, we group all non-major updates, and if the group contains packages from supported languages the columns will be there.

**Reply by rarkins** [reply]
Are you using the mergeConfidence:all-badges preset or your own enablement? Because if it's your own then it seems like it's requesting unsupported datasources (docker)

**Reply by fgreinacher** [reply]
We're using the preset. I linked a repro in the description: https://github.com/fgreinacher/renovate-digest-confidence/pull/4

**Reply by rarkins** [reply]
OK so maybe it's the grouping logic which somehow breaks it. If there's no grouping, does the node digest PR on its own have any merge confidence columns at all?

**Reply by fgreinacher** [reply]
Yes, it happens only when grouping. When it's only Docker deps these columns are missing

**Reply by rarkins** [reply]
Great, I have updated the title accordingly. I think the root cause solution here is to prevent that node row from having badge URLs, instead of changing the badge server

**Reply by fgreinacher** [reply]
Agreed, that's then my option 2 above I think :)

It works well by adapting [`prBodyDefinitions`](https://github.com/fgreinacher/renovate-digest-confidence/blob/main/renovate.json#L16-L19):

<img width="820" alt="image" src="https://github.com/renovatebot/renovate/assets/1540469/bb675821-98bf-4d3f-9220-6986269902b3">

I think it makes sense to update the default here, let me know if I should send a PR!

**Reply by rarkins** [reply]
Please do!

**Reply by HonkingGoose** [reply]
We got the PR, and merged it:

- https://github.com/renovatebot/renovate/pull/27654
